pytermgui/ansi_interface.py

```diff
@@ -107,243 +142,263 @@ class Color:
-        elif isinstance(color, int) or color.isdigit():
+        # 8-bit/Xterm colors (256 colors)
+        elif int(color) in range(256):
             color_value = f"5;{color}m"
-
         else:
             raise NotImplementedError(
                 f"Not sure what to do with {color} of type {type(color)}"
             )
+        ending = "\x1b[0m" if reset_style else ""
+        return f"\x1b[{38 + self.layer_offset};{color_value}{text}{ending}"
 
-        return (
-            f"\x1b[{38 + self.layer_offset};"
-            + color_value
-            + text
-            + (set_mode("reset") if reset_color else "")
-        )
-
-
-foreground = Color()
-"""`Color` instance to setting foreground colors"""
-
-background = Color(layer=1)
-"""`Color` instance to setting background colors"""
 
+foreground: Color = Color(0)
+"""
+`Color` instance to setting foreground colors
+"""
 
-def screen_size() -> tuple[int, int]:
-    """Get screen size using the os module
+background: Color = Color(1)
+"""
+`Color` instance to setting background colors
+"""
 
-    This is technically possible using a method of moving the
-    cursor to an impossible location, and using `report_cursor()`
-    to get where the position was clamped. This however messes
-    with the cursor position and printing gets a bit glitchy."""
 
-    try:
-        width, height = get_terminal_size()
-        return (width, height)
+class TerminalModes:
+    """
+    A class to change terminal modes
+    """
 
-    except OSError as error:
-        if error.errno in (9, 25):
-            return 0, 0
+    def __init__(self) -> None:
+        """
+        Initialize `TerminalModes` class
+        """
+        if _IS_NT:
+            self._OldStdinMode = ctypes.wintypes.DWORD()
+            self._OldStdoutMode = ctypes.wintypes.DWORD()
+            kernel32.GetConsoleMode(
+                kernel32.GetStdHandle(-10), ctypes.byref(self._OldStdinMode)
+            )
+            kernel32.GetConsoleMode(
+                kernel32.GetStdHandle(-11), ctypes.byref(self._OldStdoutMode)
+            )
+        else:
+            self._OldStdinMode = termios.tcgetattr(sys.stdin)
 
-        raise
+    @contextmanager
+    def NoEcho(self) -> None:
+        if _IS_NT:
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), 0)
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), 7)
+        else:
+            _ = self._OldStdinMode
+            _[3] = _[3] & ~(termios.ECHO | termios.ICANON)
+            termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, _)
+        try:
+            yield
+        finally:
+            if _IS_NT:
+                kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), self._OldStdinMode)
+                kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), self._OldStdoutMode)
+            else:
+                termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, self._OldStdinMode)
+
+
+terminal_modes = TerminalModes()
+"""
+TerminalModes instance
+"""
```

Comment:
More explanation would be great, though I would prefer this functionality to exist within `_Terminal` instead.